var/spack/repos/builtin/packages/ncbi-rmblastn/package.py

```diff
@@ -12,11 +12,25 @@ class NcbiRmblastn(AutotoolsPackage):
     homepage = "http://www.repeatmasker.org/RMBlast.html"
     url      = "ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.9.0/ncbi-blast-2.9.0+-src.tar.gz"
 
+    version('2.11.0', sha256='d88e1858ae7ce553545a795a2120e657a799a6d334f2a07ef0330cc3e74e1954')
     version('2.9.0', sha256='a390cc2d7a09422759fc178db84de9def822cbe485916bbb2ec0d215dacdc257')
+
+    # There is a corresponding gzipped patch file associated with each version.
+    # According to the documentation, http://www.repeatmasker.org/RMBlast.html, 
+    # Download NCBI Blast+ and rmblast patch file:
+    #   ncbi-blast-2.11.0+-src.tar.gz
+    #   isb-2.11.0+-rmblast.patch.gz
+    # The patch is downloaded and unzipped in the ncbi-rmblastn Spack package
+    # directory to make it available for the patch directive.
+    patch('isb-2.11.0+-rmblast.patch', when="@2.11.0")
```

Comment:
@glennpj This should work with something like:
```suggestion
    patch(
        'http://www.repeatmasker.org/isb-2.11.0+-rmblast.patch.gz',
        sha256='ce985abd3512834adb9ad3e4078fbf9608a33a2ee6538a1e94b641490c92f899',
        archive_sha256='0fc27781c2ea2f17645247e2f3775b5d18c56f0b62761a865347be745ea4f6be'
    )
```
but currently fails due to a bug in our fetchers when using `gunzip`. Basically we move to a directory, but gunzip always extracts the original file in-place and in the end we don't find the extracted resource where we expect it to be.

Reply:
I was able to make the patch work with the suggestion above and this diff:
```diff
diff --git a/lib/spack/spack/util/compression.py b/lib/spack/spack/util/compression.py
index 2f190e94e1..03a830109a 100644
--- a/lib/spack/spack/util/compression.py
+++ b/lib/spack/spack/util/compression.py
@@ -22,6 +22,21 @@ def allowed_archive(path):
     return any(path.endswith(t) for t in ALLOWED_ARCHIVE_TYPES)
 
 
+def _gunzip(archive_file):
+    """Like gunzip, but extracts in cwd instead of in-place.
+
+    Args:
+        archive_file (str): absolute path of the file to be decompressed
+    """
+    import gzip
+    decompressed_file = os.path.basename(archive_file.strip('.gz'))
+    working_dir = os.getcwd()
+    destination_abspath = os.path.join(working_dir, decompressed_file)
+    with gzip.open(archive_file, "rb") as f_in:
+        with open(destination_abspath, "wb") as f_out:
+            f_out.write(f_in.read())
+
+
 def decompressor_for(path, extension=None):
     """Get the appropriate decompressor for a path."""
     if ((extension and re.match(r'\.?zip$', extension)) or
@@ -30,8 +45,7 @@ def decompressor_for(path, extension=None):
         unzip.add_default_arg('-q')
         return unzip
     if extension and re.match(r'gz', extension):
-        gunzip = which('gunzip', required=True)
-        return gunzip
+        return _gunzip
     if extension and re.match(r'bz2', extension):
         bunzip2 = which('bunzip2', required=True)
         return bunzip2
```
I'll check if the approach in the diff raises any issue and get back to you. Please ping me if I don't :slightly_smiling_face:

Reply:
@alalazo Would it be prudent to merge this PR as-is and then fix it later when gzipped patches work?

Reply:
@glennpj Sure, just fix style issues and have test passing

Reply:
Arghh, trailing whitespace. I will fix that.